prov/efa/src/rdm/rxr_msg.c

```diff
@@ -111,8 +111,7 @@ int rxr_msg_select_rtm(struct rxr_ep *rxr_ep, struct rxr_op_entry *tx_entry, int
 		 * Have the remote side issue a read to copy the data instead
 		 * to work around this issue.
 		 */
-		if (tx_entry->total_len > rxr_env.shm_max_medium_size ||
-		   (tx_entry->total_len > 0 && efa_mr_is_hmem(tx_entry->desc[0])))
+		if (tx_entry->total_len >= rxr_env.shm_max_medium_size)
```

Comment:
`shm_max_medium_size` is 4k by default. I thought the optimum size for cuda memory is 2k. Right?

Reply:
That's a good point. I should also mention that based on my experiment, eager rtm + gdrcopy is faster than cudaMemcpy up to 3k.

For that we can have another threshold, e.g. in `efa_hmem_info` instead of `rxr_env.` I preferred not to do that initially for simplicity.

Do you think the additional threshold makes more sense? We can set it to 2k as you suggested as well.

Reply:
I think we should use a threshold that is optimum

Reply:
My understanding of gdrcopy is that, its performance is also affected by the CPU speed. This could make it hard to find a general optimal threshold for all platforms.

We can set the threshold to 3k based on the data on p4d for now then.